src/utils/__tests__/number.utils.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { utils } from 'ethers'
+import { NumberUtils } from '../number.utils.ts'
+import { MAX_GAS_LIMIT } from '../../constants/config.ts'
+
+const { shouldShowWrapFeeWarningModal } = NumberUtils
+
+const gasPrice = utils.parseUnits('100', 'gwei')
+const fee = gasPrice.mul(MAX_GAS_LIMIT)
+
+describe('NumberUtils', () => {
+  describe('shouldShowWrapFeeWarningModal', () => {
+    it('should return false when account balance is less than fee times multiplier', () => {
+      const amount = utils.parseEther('0.02')
+      const accountBalanceAmount = utils.parseEther('0.02')
+      const show = shouldShowWrapFeeWarningModal({ fee, amount, accountBalanceAmount })
+
+      expect(show).toEqual(false)
+    })
+
+    it('should return false when amount is less than fee times multiplier', () => {
+      const amount = utils.parseEther('0.02')
+      const accountBalanceAmount = utils.parseEther('10')
+      const show = shouldShowWrapFeeWarningModal({ fee, amount, accountBalanceAmount })
+
+      expect(show).toEqual(false)
+    })
+
+    it('should return true when amount plus fee times multiplier is greater than account balance', () => {
+      const amount = utils.parseEther('9.99')
+      const accountBalanceAmount = utils.parseEther('10')
+      const show = shouldShowWrapFeeWarningModal({ fee, amount, accountBalanceAmount })
+
+      expect(show).toEqual(true)
+    })
+
+    it('should return false when amount plus fee times multiplier is less than account balance', () => {
```

Comment:
Kinda hard for me to describe math in plain words, did do my best. If you still don't agree with wording of the comments and test descriptions, do suggest some examples.

Reply:
I'll try; but I dont understand the `if (amount.sub(multiplierDeductionFee).lte(0)) {` edge case. How do you get to it, without `amount.add(multiplierDeductionFee).gt(accountBalanceAmount)` being false

Reply:
```ts
import { expect, test } from 'vitest'
import { utils } from 'ethers'
import { NumberUtils } from '../number.utils.ts'
import { WRAP_FEE_DEDUCTION_MULTIPLIER } from '../../constants/config.ts'

const { shouldShowWrapFeeWarningModal } = NumberUtils

test('should show modal if user can not afford future transactions', () => {
  expect(shouldShowWrapFeeWarningModal({
    accountBalanceAmount: utils.parseEther('10'),
    amount: utils.parseEther('9.99'),
    fee: utils.parseEther('0.01'),
  })).toEqual(true)
})

test('should skip modal if user can already afford future transactions', () => {
  expect(shouldShowWrapFeeWarningModal({
    accountBalanceAmount: utils.parseEther('40'),
    amount: utils.parseEther('20'),
    fee: utils.parseEther('0.01'),
  })).toEqual(false)
})

test('should skip modal if user can not afford to retain fees for future transactions', () => {
  expect(shouldShowWrapFeeWarningModal({
    accountBalanceAmount: utils.parseEther('0.04'),
    amount: utils.parseEther('0.02'),
    fee: utils.parseEther('0.01'),
  })).toEqual(false)
})

test('do not recommend a negative amount / do not affect tiny amount', () => {
  // Note: this one also passes because user can already afford future transactions.
  expect(shouldShowWrapFeeWarningModal({
    accountBalanceAmount: utils.parseEther('10'),
    amount: utils.parseEther('0.005'),
    fee: utils.parseEther('0.01'),
  })).toEqual(false)

  expect(shouldShowWrapFeeWarningModal({
    accountBalanceAmount: utils.parseEther('0.01').mul(WRAP_FEE_DEDUCTION_MULTIPLIER).add(utils.parseEther('0.003')), // 0.053
    amount: utils.parseEther('0.005'),
    fee: utils.parseEther('0.01'),
  })).toEqual(false)
})
```

Reply:
This case `amount.sub(multiplierDeductionFee).lte(0)`, is for converting small amounts. Like if you would want to wROSE 0.01 ROSE -> wROSE we don't want to show modal - if that makes sense?